src/Features/Core/Portable/Diagnostics/DiagnosticHelper.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    internal static class DiagnosticHelper
```

Comment:
>DiagnosticHelper [](start = 26, length = 16)

Please create a work item tracking removal of this type when we prototype your proposal on handling dynamic diagnostic IDs.